js/services/api/guest-invites.js

```diff
@@ -112,29 +112,167 @@ const processSignUpEmailInvites = (userId, email) => {
+    .ref()
+    .child(`/potLuckGuests/${potluckId}/${userId}`)
+    .once('value')
+    .then((result) => {
+      return result.val();
+    });
+};
+
+const getGuestInviteUserDetailsAccept = (potluckId, userId) => {
+  return Promise.resolve(checkForUserInvites(potluckId, userId))
+    .then((invite) => {
+      if (invite.inviteAccepted) {
+        return userDetails.get(userId);
+      }
+    });
+};
+
+const getGuestInviteUserDetailsReject = (potluckId, userId) => {
+  return Promise.resolve(checkForUserInvites(potluckId, userId))
+    .then((invite) => {
+      if (invite.inviteDeclined) {
+        return userDetails.get(userId);
+      }
+    });
+};
+
+const getGuestInviteUserDetailsNoResponse = (potluckId, userId) => {
+  return Promise.resolve(checkForUserInvites(potluckId, userId))
+    .then((invite) => {
+      if (!invite.inviteAccepted && !invite.inviteDeclined) {
+        return userDetails.get(userId);
+      }
+    });
+};
+
+const processtPotluckUserInviteResponses = (potluckId, inviteType) => {
+  return api.get(`/potLuckGuests/${potluckId}`)
+    .then((userIds) => {
+      return Object.keys(userIds);
+    })
+    .then((result) => {
+      const promises = result
+        .map((userId) => {
+          return inviteType(potluckId, userId);
+        });
+
+      return Promise.all(promises);
+    })
+    .then((users) => {
+      return users.filter(user => !!user);
+    })
+    .catch(() => {
+      return [];
+    });
+};
+
+const getPotluckGuestInviteEmails = (potluckId) => {
+  return firebase.database()
+    .ref()
+    .child(`potLuckInvites/${potluckId}/guests`)
+    .once('value')
+    .then((result) => {
+      return result.val();
+    });
+};
+
+const processPotluckEmailInviteNoResponses = (potluckId) => {
+  return Promise.resolve(getPotluckGuestInviteEmails(potluckId))
+    .then((result) => {
+      const emails = Object.keys(result).map((email) => {
+        return atob(email);
+      });
+
+      return emails;
+    })
+    .catch(() => {
+      return [];
+    });
+};
+
+const createGuestResponses = (inviteResponses) => {
+  const guestResponses = [];
+
+  Object.keys(inviteResponses).map((key) => {
+    if (key === 'userAccept' && inviteResponses[key].length > 0) {
+      inviteResponses[key].forEach((item) => {
+        guestResponses.push([`${item.firstName} ${item.lastName}`, '+']);
+      });
+    }
+    if (key === 'userReject' && inviteResponses[key].length > 0) {
+      inviteResponses[key].forEach((item) => {
+        guestResponses.push([`${item.firstName} ${item.lastName}`, '-']);
+      });
+    }
+    if (key === 'userNoResponse' && inviteResponses[key].length > 0) {
+      inviteResponses[key].forEach((item) => {
+        guestResponses.push([`${item.firstName} ${item.lastName}`, '?']);
+      });
+    }
+    if (key === 'emailNoResponse' && inviteResponses[key].length > 0) {
+      inviteResponses[key].forEach((item) => {
+        guestResponses.push([item, '?']);
+      });
+    }
```

Comment:
Suggest a refactor to write a function:

```js
const createGuestResponse = (key, item) => {
  const responseMap = {
    userAccept: [`${item.firstName} ${item.lastName}`, '+'],
    userReject: [`${item.firstName} ${item.lastName}`, '-'],
    ...etc
  }

  return responseMap[key];
}
```

And just throw in a guard clause to handle the `inviteResponses[key].length > 0` repetition.